### ISSUE TITLE
Tile Update Error

### ISSUE THREAD

**Issue by hahaharry10** [hahaharry10/ConwaysGameOfLife#12]
With current properties:

```C
#define SCREEN_WIDTH 1940
#define SCREEN_HEIGHT 1080
#define GRID_START_X 0
#define GRID_START_Y 40

#define CELL_WIDTH 10
#define CELL_HEIGHT 10
#define CELL_GRID_WIDTH ((SCREEN_WIDTH - GRID_START_X) / CELL_WIDTH)
#define CELL_GRID_HEIGHT ((SCREEN_HEIGHT - GRID_START_Y) / CELL_HEIGHT)
```

around column 103, the grid does not update correctly.

Look at the following screen recording:

https://github.com/user-attachments/assets/eeca4765-5257-4912-bb9e-00ba7aa605b5

And the corresponding keylog:
<img width="348" alt="Screenshot 2024-08-30 at 13 28 38" src="https://github.com/user-attachments/assets/9d1a42c2-5a8f-4dd7-9f6a-d5caaf1e8c6c">

The generation is simulated incorrectly creating this incorrect output.

**Comment by hahaharry10** [comment]
For input:
<img width="388" alt="Input Log" src="https://github.com/user-attachments/assets/0e3b677b-b3fc-4923-a1b6-b68b5bf9830e">

The calculated neighbours are the following:

<img width="350" alt="Neighbour Count" src="https://github.com/user-attachments/assets/f9e06c74-7071-4b56-b5ca-bc614ca9864f">

The correct neighbour count is actually:

```
1 2 3 2 1 0
1 2 4 4 3 1
1 3 4 4 2 1
0 1 2 3 2 1
```

So, the second-to-last count on the bottom line and the second-to-last count on the penultimate line are incorrectly calculated (calculated as 1 but should be 2). 

I suspect the issue is with the logic and cell-checking of the upper-left diagonal neighbour.

**Comment by hahaharry10** [comment]
fixed in line 37 of commit 8651be6
